standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -546,7 +553,11 @@ public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Tab
       }
     }
 
-    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER)) {
+    String statsSource = Optional.ofNullable(envContext).orElse(new EnvironmentContext(Collections.emptyMap()))
+            .getProperties().getOrDefault(HIVE_ICEBERG_STATS_SOURCE, HiveMetaHook.ICEBERG.toLowerCase()).toLowerCase();
+    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER) && 
+        !(isIcebergTable && statsSource.equals(HiveMetaHook.ICEBERG.toLowerCase()))) {
+      LOG.debug(STATS_CALC_CALL_LOG_MSG + tbl.getCatName() + "." + tbl.getDbName() + "." + tbl.getTableName());
```

Comment:
Use logger format
```
LOG.debug(STATS_CALC_CALL_LOG_MSG {}.{}.{}, tbl.getCatName(), tbl.getDbName(), tbl.getTableName()));
```
I don't think the constant STATS_CALC_CALL_LOG_MSG is required, can directly put the text over here

Reply:
Done.